crates/cli/tests/module_generator/mod.rs

```diff
@@ -0,0 +1,240 @@
+        self.next_func_index += 1;
+    }
+
+    pub fn assign_start_fn(&mut self) {
+        self.start_fn = Some(self.next_func_index);
+        self.next_func_index += 1;
+    }
+
+    pub fn assign_javy_quickjs_provider_memory(&mut self) {
+        self.javy_quickjs_provider_memory = Some(self.next_memory_index);
+        self.next_memory_index += 1;
+    }
+
+    pub fn assign_memory(&mut self) {
+        self.memory = Some(self.next_memory_index);
+        self.next_memory_index += 1;
+    }
+}
+
+fn add_types(module: &mut Module, indices: &mut Indices) {
+    let mut types = TypeSection::new();
+
+    // canonical_abi_realloc
+    types.function(
+        vec![ValType::I32, ValType::I32, ValType::I32, ValType::I32],
+        vec![ValType::I32],
+    );
+    indices.assign_realloc_ty();
+
+    // eval_bytecode
+    types.function(vec![ValType::I32, ValType::I32], vec![]);
+    indices.assign_eval_bytecode_ty();
+
+    // _start
+    types.function(vec![], vec![]);
+    indices.assign_start_ty();
+
+    module.section(&types);
+}
+
+fn add_imports(module: &mut Module, indices: &mut Indices) {
+    const IMPORT_NAMESPACE: &str = "javy_quickjs_provider_v1";
+    let mut imports = ImportSection::new();
+
+    imports.import(
+        IMPORT_NAMESPACE,
+        "canonical_abi_realloc",
+        EntityType::Function(indices.realloc_ty.unwrap()),
+    );
+    indices.assign_realloc_fn();
+
+    imports.import(
+        IMPORT_NAMESPACE,
+        "eval_bytecode",
+        EntityType::Function(indices.eval_bytecode_ty.unwrap()),
+    );
+    indices.assign_eval_bytecode_fn();
+
+    imports.import(
+        IMPORT_NAMESPACE,
+        "memory",
+        EntityType::Memory(MemoryType {
+            minimum: 0,
+            maximum: None,
+            memory64: false,
+            shared: false,
+        }),
+    );
+    indices.assign_javy_quickjs_provider_memory();
+
+    module.section(&imports);
+}
+
+fn add_functions(module: &mut Module, indices: &mut Indices) {
+    let mut functions = FunctionSection::new();
+    functions.function(indices.start_ty.unwrap());
+    indices.assign_start_fn();
+    module.section(&functions);
+}
+
+fn add_memories(module: &mut Module, indices: &mut Indices) {
+    let mut memories = MemorySection::new();
+    memories.memory(MemoryType {
+        minimum: 0,
+        maximum: None,
+        memory64: false,
+        shared: false,
+    });
+    indices.assign_memory();
+    module.section(&memories);
+}
+
+fn add_exports(module: &mut Module, indices: &Indices) {
+    let mut exports = ExportSection::new();
+    exports.export("memory", ExportKind::Memory, indices.memory.unwrap());
+    exports.export("_start", ExportKind::Func, indices.start_fn.unwrap());
+    module.section(&exports);
+}
+
+fn add_data_count(module: &mut Module, count: u32) {
+    module.section(&DataCountSection { count });
+}
+
+fn add_code(module: &mut Module, indices: &Indices, bytecode_len: i32) {
```

Comment:
Can we add an internal comment that includes that wat that this function is generating? It will make it more manageable for newcomers.

Reply:
Actually it might be better to add this comment as part of `generate_module` function at the top.